substrate/primitives/runtime/src/lib.rs

```diff
@@ -1196,6 +1225,39 @@ mod tests {
 		assert!(multi_sig.verify(msg, &multi_signer.into_account()));
 	}
 
+	#[test]
+	fn multi_signature_eth_verify_works() {
+		let msg = &b"test-message"[..];
+		let (pair, _) = ecdsa::Pair::generate();
+
+		let signature = pair.sign(&msg);
+		assert!(ecdsa::Pair::verify(&signature, msg, &pair.public()));
+
+		let multi_sig = MultiSignature::Eth(signature);
+		let multi_signer = MultiSigner::Eth(pair.public());
+		assert!(multi_sig.verify(msg, &multi_signer.into_account()));
+
+		let multi_signer = MultiSigner::Eth(pair.public());
+		assert!(multi_sig.verify(msg, &multi_signer.into_account()));
```

Comment:
Why are you doing this twice?

Reply:
Copy paste from another test. I fixed both now.